### PR TITLE
Add shared key generation

### PR DESCRIPTION
**Pull request by verificatorrus**
This function returns base58 encoded shared key from base58 encoded А private & B public keys for enhancement of the application as described in: https://github.com/wavesplatform/curve25519-js
The result should not be used directly as a key, but should be processed with a one-way function (e.g. HSalsa20 as in NaCl, or any secure cryptographic hash function, such as SHA-256, or key derivation function, such as HKDF).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/wavesplatform/wavesplatform.core.js"
+    "url": "https://github.com/gvoio/wavesplatform.core.js.git"
```
Comment (xenohunter) [inline package.json]: Please, unwind the _package.json_ changes.

File: package.json [hunk 2]
```diff
@@ -32,5 +32,8 @@
   },
   "scripts": {
     "postinstall": "bower install"
+  },
+  "dependencies": {
+    "jasmine-core": "^2.99.1"
```
Comment (xenohunter) [inline package.json]: And also remove this dependency.

**Comment by verificatorrus** [comment]
Thanks for the prompt answer. I have made the requested changes.

**Comment by verificatorrus** [comment]
Thanks for the remark. I corrected this mistake. Also made the function prepareKey public, for the possibility of preparing a sharedKey before using.

**Comment by xenohunter** [comment]
Thank you again for the contribution!

**Comment by xenohunter** [comment]
I also built it and committed the changes.

**Pull request merged**